src/eve/codegen.py

```diff
@@ -148,7 +150,8 @@ def format_cpp_source(
         sort_includes: bool = False,
     ) -> str:
         """Format C++ source code using clang-format."""
-        args = ["clang-format"]
+        assert isinstance(_CLANG_FORMAT_EXECUTABLE, str)
```

Comment:
Ah ha! Is this what made mypy stop complaining about the types in the list argument to `Popen`? @egparedes

Reply:
yes, mypy understands `assert`s and I think also some simple if conditions